src/tools/sfdk/dbus.cpp

```diff
@@ -0,0 +1,158 @@
+/****************************************************************************
+**
+** Copyright (C) 2020 Jolla Ltd.
+** Contact: http://jolla.com/
+**
+** This file is part of Qt Creator.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** Other Usage
+**
+** Alternatively, this file may be used in accordance with the terms and
+** conditions contained in a signed written agreement between you and Digia.
+**
+****************************************************************************/
+
+#include "dbus.h"
+
+#include <QDBusAbstractAdaptor>
+#include <QDBusConnection>
+#include <QDBusMessage>
+
+#include <utils/qtcassert.h>
+
+#include "remoteprocess.h"
+#include "sdkmanager.h"
+#include "sfdkglobal.h"
+
+using namespace Sfdk;
+
+namespace Sfdk {
+
+namespace {
+const char DBUS_SERVICE_NAME_TEMPLATE[] = "org.sailfishos.sfdk.I%1";
+const char DBUS_SFDK_ERROR[] = "org.sailfishos.sfdk.Error";
+const char DBUS_CONFIGURED_DEVICE_PATH[] = "/device";
+} // namespace anonymous
+
+class AbstractAdaptor : public QDBusAbstractAdaptor
```

Comment:
I somewhat dislike the name of the class AbstractAdaptor. It's derived from QDBusAbstractAdaptor, and it's less abstract than QDBusAbstractAdaptor. But the name implies the opposite! Well, I don't have any better naming suggestion, so feel free to ignore this comment ;)